File: app/src/main/java/com/gnardini/tvshowcontroller/networking/ControllerService.java
package com.gnardini.tvshowcontroller.networking;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ControllerService {

    @GET("/fetch")
    Call<List<String>> fetch(@Query("directory") String directory);

    @GET("/maximize")
    Call<String> maximize();

    @GET("/pause")
    Call<String> playOrPause();

    @GET("/volume")
    Call<String> setVolume(@Query("direction") String direction);

    @GET("/rewind")
    Call<String> rewind();

    @GET("/fast_forward")
    Call<String> fastForward();

}

File: app/src/main/java/com/gnardini/tvshowcontroller/networking/ControllerService.kt
package com.gnardini.tvshowcontroller.networking

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ControllerService {

    @GET("/fetch")
    fun fetch(@Query("directory") directory: String): Call<List<String>>

    @GET("/maximize")
    fun maximize(): Call<String>

    @GET("/pause")
    fun playOrPause(): Call<String>

    @GET("/volume")
    fun setVolume(@Query("direction") direction: String): Call<String>

    @GET("/rewind")
    fun rewind(): Call<String>

    @GET("/fast_forward")
    fun fastForward(): Call<String>

}
